packages/promise/src/PromiseCacheProvider.tsx

```diff
@@ -0,0 +1,12 @@
+import { type PropsWithChildren } from 'react'
+import { PromiseCacheContext } from './contexts'
+import type { PromiseCache } from './PromiseCache'
+
+type PromiseCacheProviderProps = PropsWithChildren<{ cache: PromiseCache }>
+
+/**
+ * @experimental This is experimental feature.
+ */
+export const PromiseCacheProvider = ({ cache, children }: PromiseCacheProviderProps) => (
+  <PromiseCacheContext.Provider value={cache}>{children}</PromiseCacheContext.Provider>
+)
```

Comment:
Here is the implementation reflecting the review from https://github.com/toss/suspensive/pull/1066#discussion_r1672283896